### DISCUSSION TITLE
Shelly 2.5 - dual power metering for totals

### DISCUSSION THREAD

**Discussion by sVnsation** [arendst/Tasmota#13030] [Ideas]
Refering to issue #6160 there still is the demand to also split the totals for dual power metering.
Any plans to add this feature?

**Comment by ascillato** [comment]
Feature added to Tasmota

https://github.com/arendst/Tasmota/commit/d3fc03d3980c25cd8b66063405bcab12adadaff5

**Comment by sVnsation** [comment]
Seems to work fine on the first sight. 

Maybe a note for others, that are testing: 
Data only saves to flash on occasion to minimize flash writes. A restart will save volatile data.

What i noticed so far on my quick test:
After a fresh power up energy data shows 0. A switch toggle is needed to show energy data.
Transfer of energy_total_0 and energy_total_1 to home assistant does not seem to work: state unknown.

**Reply by sfromis** [reply]
Both slow flash updates (midnight and when other things cause a save) and no energy data while power is off (unless activating `SetOption21`) are entirely regular parts of how the energy monitoring support in Tasmota works.

**Comment by sVnsation** [comment]
The Values for Total Energy are send via MQTT.
But hatasmota throws exception with the new Setoption129. Opened a Issue:   https://github.com/emontnemery/hatasmota/issues/120

In the mean time i switched to deprecated MQTT Discovery via SetOption19. Here everything is working as expected.
